include/cuco/reduction_functors.cuh

```diff
@@ -135,8 +135,25 @@ class reduction_functor : detail::reduction_functor_base {
                                 Func,
                                 cuda::atomic<value_type, cuda::thread_scope_thread>&,
                                 value_type>;
+  static constexpr bool atomic_const_invocable_ =
+    cuda::std::is_invocable_r_v<value_type,
+                                Func,
+                                cuda::atomic<value_type, cuda::thread_scope_system> const&,
+                                value_type> ||
+    cuda::std::is_invocable_r_v<value_type,
+                                Func,
+                                cuda::atomic<value_type, cuda::thread_scope_device> const&,
+                                value_type> ||
+    cuda::std::is_invocable_r_v<value_type,
+                                Func,
+                                cuda::atomic<value_type, cuda::thread_scope_block> const&,
+                                value_type> ||
+    cuda::std::is_invocable_r_v<value_type,
+                                Func,
+                                cuda::atomic<value_type, cuda::thread_scope_thread> const&,
+                                value_type>;
```

Comment:
Listing all possible values for `cuda::thread_scope` is very ugly. Does anyone have a better solution?